### PR TITLE
feat(hamburger): make hamburger controlled

### LINKED ISSUE #1164: Lukkekryss/hamburger bugger
Når man går inn og ut av meny / innholdsside bytter hamburgerikon og lukkekryss slik at hamburger vises i meny når det skulle vært kryss og tilsvarende på innholdssidene vises lukkekryss når hamburgermeny skal vises

### PR DESCRIPTION
**Pull request by joms**
affects: @fremtind/jkl-hamburger-react, @fremtind/jkl-hamburger

in stead of being uncontrolled, the component is simplified and all
logic is moved outside the component. this is to fix issues where
external factors may have to alter the state of the hamburger component

BREAKING CHANGE:
API is changed to be controlled

Closes #1164 

## ☑️ Submission checklist

-   [X] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [X] `yarn build` works locally with my changes
-   [X] I have added tests that prove my fix is effective or that my feature works
-   [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/hamburger-react/README.md [hunk 1]
```diff
@@ -25,15 +25,17 @@ import "@fremtind/jkl-hamburger/hamburger.min.css";
...
 ```
 
 Ved bruk på mørk bakgrunn brukes prop-en `negative` for å invertere ikonene.
 
 ```jsx
 <div style={{ backgroundColor: "black" }}>
-    <Hamburger negative />
+    <Hamburger negative  onClick={() => setIsMenuOpen(!isMenuOpen)} isOpen={isMenuOpen} />
 </div>
 ```
```
Comment (piofinn) [inline packages/hamburger-react/README.md:41]: Kanskje bare ta vekk dette, siden `negative`/`inverted` er deprecated?

File: packages/hamburger-react/documentation/Example.tsx [hunk 2]
```diff
@@ -3,18 +3,14 @@ import { ExampleComponentProps } from "@fremtind/jkl-portal-components";
...
 
 const Example = ({ boolValues }: ExampleComponentProps) => {
-    const [clickOutside, toggleClickOutside] = useState(false);
-    const toggleState = (enable: boolean) => toggleClickOutside(enable);
+    const [isOpen, setIsOpen] = useState(false);
 
-    const divRef = React.useRef(null);
     return (
-        <div style={{ padding: "20px" }} ref={divRef}>
+        <div style={{ padding: "20px" }}>
```
Comment (piofinn) [inline packages/hamburger-react/documentation/Example.tsx:9]: Denne div'en trengs vel ikke nå?

File: portal/src/components/Header/components/FullScreenMenu.tsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ import { useFullscreenMenu } from "../../../contexts/fullscreenMenuContext";
 import { useFullScreenMenuAnimaiton } from "./useFullScreenMenuAnimation";
 
 import "./FullScreenMenu.scss";
+import { Hamburger } from "../../../../../packages/hamburger-react/src";
```
Comment (piofinn) [inline portal/src/components/Header/components/FullScreenMenu.tsx:13]: Autoimport FTW 🙃

```suggestion
import { Hamburger } from "@fremtind/jkl-hamburger-react";
```

**Review by piofinn — CHANGES_REQUESTED** [review]
En liten autoimport-feil som bør fikses. De to andre kommentarene har ikke så mye å si

File: packages/hamburger-react/README.md (hunk 1 above)
Comment (joms) [reply packages/hamburger-react/README.md:41]: Ja, det gir forsåvidt mening

**Comment by joms** [comment]
> En liten autoimport-feil som bør fikses. De to andre kommentarene har ikke så mye å si

Fiksa alt jeg <3

**Review by piofinn — APPROVED** [review]

**Pull request merged**